src/neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -60,16 +61,39 @@ private void DistributeGas(ApplicationEngine engine, UInt160 account, NeoAccount
             // PersistingBlock is null when running under the debugger
             if (engine.Snapshot.PersistingBlock == null) return;
 
-            BigInteger gas = CalculateBonus(engine.Snapshot, state.Balance, state.BalanceHeight, engine.Snapshot.PersistingBlock.Index);
+            BigInteger gas = CalculateBonus(engine.Snapshot, state.VoteTo, state.Balance, state.BalanceHeight, engine.Snapshot.PersistingBlock.Index);
             state.BalanceHeight = engine.Snapshot.PersistingBlock.Index;
             GAS.Mint(engine, account, gas);
         }
 
-        private BigInteger CalculateBonus(StoreView snapshot, BigInteger value, uint start, uint end)
+        private BigInteger CalculateBonus(StoreView snapshot, ECPoint vote, BigInteger value, uint start, uint end)
         {
             if (value.IsZero || start >= end) return BigInteger.Zero;
             if (value.Sign < 0) throw new ArgumentOutOfRangeException(nameof(value));
 
+            BigInteger neoHolderReward = CalculateNeoHolderReward(snapshot, value, start, end);
+            if (vote is null) return neoHolderReward;
+
+            BigInteger startRewardPerNeo, endRewardPerNeo;
+            StorageKey keyStart = CreateStorageKey(Prefix_VoterRewardPerCommittee).Add(vote).AddBigEndian(uint.MaxValue - end);
+            StorageKey keyEnd = CreateStorageKey(Prefix_VoterRewardPerCommittee).Add(vote).AddBigEndian(uint.MaxValue - start);
+            using (var enumerator = snapshot.Storages.FindRange(keyStart, keyEnd).GetEnumerator())
+            {
+                if (!enumerator.MoveNext()) return neoHolderReward;
+                endRewardPerNeo = enumerator.Current.Value;
+            }
+            using (var enumerator = snapshot.Storages.Seek(keyEnd.ToArray(), IO.Caching.SeekDirection.Backward).GetEnumerator())
```

Comment:
Should it use `SeekDirection.Forward`?

Reply:
1. `SeekDirection.Forward` which means that user can receive `voter` reward, as long as he/she vote before the end of epoch.
2. `SeekDirection.Backward` which means that user can only receive `voter` reward after a full epoch.

Which do you prefer?

Reply:
I think it should be `Forward`. Otherwise, if `start` and `end` are in the same epoch, the reward may become negative.

Reply:
@Tommo-L

Reply:
Got, We should use `Forward`, then we don't have this problem -- lost `voter` reward in epoch.